evergreen/set-up-env.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```

Comment:
Compared with `set-up-env-centos7.sh`, this file just made setting CC and CXX optional. Not sure why Github thought this is a rewrite.